core/storage/app_feedback_report/gae_models.py

```diff
@@ -0,0 +1,591 @@
+    daily_ticket_stats = datastore_services.JsonProperty(
+        required=True, indexed=False)
+    # The schema version for parameter statistics in this entity.
+    daily_ticket_stats_schema_version = datastore_services.IntegerProperty(
+        required=True, indexed=True)
+
+    @classmethod
+    def create(
+            cls, platform, ticket_id, stats_tracking_date, daily_ticket_stats):
+        """Creates a new AppFeedbackReportStatsModel instance and returns its
+        ID.
+
+        Args:
+            ticket_id: str. The ID for the ticket these stats aggregate on.
+            platform: str. The platform the stats are aggregating for.
+            stats_tracking_date: datetime.date. The date in UTC that this entity
+                is tracking stats for.
+            daily_ticket_stats: dict. The daily stats for this entity, keyed
+                by the parameter witch each value mapping a parameter value to
+                the number of reports that satisfy that parameter value.
+
+        Returns:
+            AppFeedbackReportStatsModel. The newly created
+            AppFeedbackReportStatsModel instance.
+        """
+        entity_id = cls._generate_id(platform, ticket_id, stats_tracking_date)
+        stats_entity = cls(
+            id=entity_id, ticket_id=ticket_id, platform=platform,
+            stats_tracking_date=stats_tracking_date,
+            daily_ticket_stats=daily_ticket_stats,
+            daily_ticket_stats_schema_version=(
+                feconf.CURRENT_REPORT_STATS_SCHEMA_VERSION))
+        stats_entity.update_timestamps()
+        stats_entity.put()
+        return entity_id
+
+    @classmethod
+    def _generate_id(cls, platform, ticket_id, stats_tracking_date):
+        """Generates key for the instance of AppFeedbackReportStatsModel
+        class in the required format with the arguments provided.
+
+        Args:
+            platform: str. The platform this entity is aggregating on.
+            ticket_id: str. The ID for the ticket these stats aggregate on.
+            stats_tracking_date: date. The date these stats are tracking on.
+
+        Returns:
+            str. The generated ID for this entity of the form
+            '[platform]:[ticket_id]:[date.seconds]'.
+        """
+        for _ in python_utils.RANGE(base_models.MAX_RETRIES):
+            new_id = '%s:%s:%s' % (
+                platform, ticket_id, stats_tracking_date.isoformat())
+            if not cls.get_by_id(new_id):
+                return new_id
+        raise Exception(
+            'The id generator for AppFeedbackReportStatsModel is producing too '
+            'many collisions.')
+
+    @classmethod
+    def get_stats_for_ticket(cls, ticket_id):
+        """Fetches the stats for a single ticket.
+
+        Args:
+            ticket_id: str. The id of the ticket to get stats for.
+
+        Returns:
+            list(AppFeedbackReportStatsModel). A list of IDs for the stats of
```

Comment:
Is this comment correct? It seems like a list of models is being returned, not IDs.

Reply:
Ah, thanks for catching, should be list(str), updated.